weaver/main.go

```diff
@@ -121,12 +129,23 @@ func main() {
 
 	router := weave.NewRouter(config, name, nickName)
 	log.Println("Our name is", router.Ourself)
-	router.Start()
-	initiateConnections(router, peers)
+	var allocator *ipam.Allocator
 	if httpAddr != "" {
-		go handleHTTP(router, httpAddr)
+		if iprangeCIDR != "" {
+			allocator = createAllocator(router, apiPath, iprangeCIDR, determineQuorum(peerCount, peers))
+		} else {
+			if peerCount > 0 {
+				log.Fatal("-initpeercount flag specified without -iprange")
+			}
+		}
+	}
+	if httpAddr == "" || iprangeCIDR == "" {
+		router.NewGossip("IPallocation", &ipam.DummyAllocator{})
```

Comment:
This is frightfully convoluted, and wrong...

- we shouldn't call `handleHTTP` (in the code following this) unless `httpAddr != nil`
- we should create the allocator regardless of whether there is an `httpAddr` present. Otherwise quorum may not be reached.

So I think the whole lot should just be
````
	var allocator *ipam.Allocator
	if ipRangeCIDR != "" {
		allocator = createAllocator(router, apiPath, iprangeCIDR, determineQuorum(peerCount, peers))
	} else if peerCount > 0 {
		log.Fatal("-initpeercount flag specified without -iprange")
	} else {
		router.NewGossip("IPallocation", &ipam.DummyAllocator{})
	}
	router.Start()
	initiateConnections(router, peers)
 	if httpAddr != "" {
		go handleHTTP(router, httpAddr)
	}